xds/internal/balancer/edsbalancer/eds_impl.go

```diff
@@ -169,6 +169,9 @@ func (edsImpl *edsBalancerImpl) updateDrops(dropPolicies []xdsclient.OverloadDro
 			dropsChanged = true
 		}
 	}
+	if len(edsImpl.drops) != len(newDrops) {
```

Comment:
Would it better to just create the `newDrops` slice in the above loop and do a cmp.Equal() on the existing drops `edsImpl.drops`, instead of trying to track at every possible stage as to whether the drops config has changed? If required we can implement an `Equal` method on the `dropper` type, or pass appropriate options to include unexported fields to `cmp.Equal`. Wdyt?

Reply:
This would definitely look cleaner. And I made #3434.

This fix will be cherry picked into the release branch, and the other change is fairly big (even though pretty straight forward). Let's merge this and cherry pick, and do that on master later? What do you think?